source/jormungandr/jormungandr/utils.py

```diff
@@ -150,21 +152,22 @@ def timestamp_to_datetime(timestamp, tz=None):
 
     dt = datetime.utcfromtimestamp(timestamp)
 
-    timezone = tz or get_timezone()
+    timezone = tz or get_timezone(req_id or request.id)
     if timezone:
         dt = pytz.utc.localize(dt)
         return dt.astimezone(timezone)
     return None
 
 
-def dt_to_str(dt):
+@functools32.lru_cache(2048)
+def dt_to_str(dt, req_id):
     return dt.strftime(DATETIME_FORMAT)
 
-
-def timestamp_to_str(timestamp):
-    dt = timestamp_to_datetime(timestamp)
+@functools32.lru_cache(2048)
```

Comment:
:+1: 

But :) how did you come up with 2048 ? Do you have any info about cache miss/hit rate ?

Reply:
it comes totally from nowhere :D I'll do more test on it

Reply:
About the hit/miss, it's about ~400000 hits and ~2000 misses